bin/audit-svg-colors.js

```diff
@@ -0,0 +1,282 @@
+ */
+
+/**
+ * External dependencies
+ */
+
+const { execSync } = require( 'child_process' );
+const path = require( 'path' );
+const { readFileSync } = require( 'fs' );
+
+const _ = require( 'lodash' );
+const chroma = require( 'chroma-js' );
+const PALETTE = require( '@automattic/color-studio' );
+
+/**
+ * Palette color subsets
+ */
+
+// The subset of palette colors allowed in illustrations
+const PALETTE_ILLUSTRATION_COLORS = _.pickBy( PALETTE.colors, ( colorValue, colorName ) => {
+	// Avoid using pure black
+	if ( colorValue === '#000' ) {
+		return;
+	}
+	// Since illustrations are a part of WordPress.com’s and Jetpack’s visual
+	// identity, we prefer them over generic Blue, Green, and Celadon. We don’t
+	// use WooCommerce Purple either
+	return ! (
+		_.startsWith( colorName, 'Blue' ) ||
+		_.startsWith( colorName, 'Green' ) ||
+		_.startsWith( colorName, 'Celadon' ) ||
+		_.startsWith( colorName, 'WooCommerce Purple' )
+	);
+} );
+
+// The subset of palette colors used in app-related images is slightly wider
+// than what we allow for in illustration use (the above)
+const PALETTE_APP_COLORS = _.pickBy( PALETTE.colors, ( colorValue, colorName ) => {
+	// Avoid using pure black
+	if ( colorValue === '#000' ) {
+		return;
+	}
+	// Don’t use WooCommerce Purple for any WordPress.com images
+	return ! _.startsWith( colorName, 'WooCommerce Purple' );
+} );
+
+// Making sure both sets contain only unique color values
+// (the palette defines aliases for some colors)
+const PALETTE_ILLUSTRATION_COLOR_VALUES = _.uniq( Object.values( PALETTE_ILLUSTRATION_COLORS ) );
+const PALETTE_APP_COLOR_VALUES = _.uniq( Object.values( PALETTE_APP_COLORS ) );
+
+/**
+ * SVG image rules
+ */
+
+// The image paths that match the following patterns will not be processed
+const SVG_IGNORE_PATHS = [
+	// Common logos found in the repository
+	/(?:google-photos|paypal|stripe)(-logo)?\.svg$/,
+
+	// Selected logos from the G-Suite asset directory
+	/images\/g-suite\/logo_/,
+
+	// Credit card and payment gateway logos (the disabled versions are allowed)
+	/upgrades\/cc-(?:amex|diners|discover|jcb|mastercard|unionpay|visa)\.svg$/,
+	/upgrades\/(?:alipay|bancontact|brazil-tef|emergent-paywall|eps|giropay|ideal|netbanking|p24|paypal|paytm|sofort|tef|wechat)\.svg$/,
+
+	// Old WooCommerce mascotte
+	/ninja-joy\.svg$/,
+
+	// Specific directories
+	/^docs/,
+	/^static\/images\/marketing/,
+	/^static\/images\/me/,
+];
+
+// The image paths that match the following patterns will use `PALETTE_APP_COLORS`,
+// while all other paths with fall back to `PALETTE_ILLUSTRATION_COLORS`
+const SVG_APP_PATHS = [
+	// Color scheme thumbnails
+	/color-scheme-thumbnail-[a-z-]+\.svg$/,
+
+	// Gutenberg images
+	/^static\/images\/illustrations\/gutenberg/,
+
+	// Plan icons
+	/^static\/images\/plans\//,
+];
+
+// The regular expression used to identify color values
+const SVG_VALUE_EXPRESSION = /(?:fill|flood-color|lighting-color|stop-color|stroke)="([a-z0-9#]*?)"/gi;
```

Comment:
If this is intended to target hex values, we could tighten it up a bit.

There are more valid values than hex that we'll find here. That's probably acceptable for a tool like this.

Reply:
This also captures explicit color names like `white`, `black`, etc.

I’m leaving out `rgba()` and other similar formats as I’ve never seen them using within a SVG file.